opencti-platform/opencti-front/src/private/components/analysis/Opinions.tsx

```diff
@@ -0,0 +1,224 @@
+  const {
+    numberOfElements,
+    filters,
+    searchTerm,
+    sortBy,
+    orderAsc,
+    openExports,
+  } = viewStorage;
+  const {
+    selectedElements,
+    deSelectedElements,
+    selectAll,
+    handleClearSelectedElements,
+    handleToggleSelectAll,
+    onToggleEntity,
+    numberOfSelectedElements,
+  } = useEntityToggle<OpinionLine_node$data>('view-opinions');
+
+  const queryRef = useQueryLoading<OpinionsLinesPaginationQuery>(
+    opinionsLinesQuery,
+    paginationOptions,
+  );
+  const renderLines = () => {
+    let exportContext = null;
+    if (objectId) {
+      exportContext = `of-entity-${objectId}`;
+    } else if (authorId) {
+      exportContext = `of-entity-${authorId}`;
+    }
+    let finalFilters = filters;
+    finalFilters = {
+      ...finalFilters,
+      entity_type: [{ id: 'Opinion', value: 'Opinion' }],
+    };
+    const isRuntimeSort = helper?.isRuntimeFieldEnable() ?? false;
+    const dataColumns = {
+      opinion: {
+        label: 'Opinion',
+        width: '35%',
+        isSortable: true,
+      },
+      createdBy: {
+        label: 'Author',
+        width: '12%',
+        isSortable: isRuntimeSort,
+      },
+      creator: {
+        label: 'Creators',
+        width: '12%',
+        isSortable: isRuntimeSort,
+      },
+      objectLabel: {
+        label: 'Labels',
+        width: '15%',
+        isSortable: false,
+      },
+      created: {
+        label: 'Date',
+        width: '10%',
+        isSortable: true,
+      },
+      x_opencti_workflow_id: {
+        label: 'Status',
+        width: '8%',
+        isSortable: true,
+      },
+      objectMarking: {
+        label: 'Marking',
+        isSortable: isRuntimeSort,
+        width: '8%',
+      },
+    };
+    return (
+      <div>
+        <ListLines
+          sortBy={sortBy}
+          orderAsc={orderAsc}
+          dataColumns={dataColumns}
+          handleSort={storageHelpers.handleSort}
+          handleSearch={storageHelpers.handleSearch}
+          handleAddFilter={storageHelpers.handleAddFilter}
+          handleRemoveFilter={storageHelpers.handleRemoveFilter}
+          handleToggleExports={storageHelpers.handleToggleExports}
+          handleToggleSelectAll={handleToggleSelectAll}
+          selectAll={selectAll}
+          openExports={openExports}
+          noPadding={typeof onChangeOpenExports === 'function'}
+          exportEntityType="Opinion"
+          exportContext={exportContext}
+          keyword={searchTerm}
+          filters={filters}
+          paginationOptions={paginationOptions}
+          numberOfElements={numberOfElements}
+          iconExtension={true}
+          availableFilterKeys={[
+            'x_opencti_workflow_id',
+            'labelledBy',
+            'createdBy',
+            'creator',
+            'markedBy',
+            'confidence',
+            'created_start_date',
+            'created_end_date',
+          ]}
+        >
+          {queryRef && (
+            <QueryRenderer
```

Comment:
A queryRef should not be mixed with a queryRenderer. Tell me if you need some insight about how we use relay

Reply:
Fixed